### DISCUSSION TITLE
Are .Net 6.0 dll-s distributed in R2R format?

### DISCUSSION THREAD

**Discussion by iSazonov** [dotnet/runtime#60781] [Q&A] [answered]
1. Are .Net 6.0 dll-s distributed in R2R format?
2. Which utility I could use to quickly determine the dll format? Or set an environment variable to show trace?

I compiled PowerShell with .Net 6.0 RC1 and see some .Net dll-s in R2R but not System.IO.FileSystem.Primitives.dll (as example) as IlSpy shows.

**Comment by EgorBo** [comment, marked as answer]
> Are .Net 6.0 dll-s distributed in R2R format?

Yes

> Which utility I could use to quickly determine the dll format? Or set an environment variable to show trace?

I personally use [r2rdump](https://github.com/dotnet/runtime/tree/main/src/coreclr/tools/r2rdump) utility built locally from dotnet/runtime. Perhaps, there is an easier way to check a dll if it contains R2R image embedded (ildasm?)

**Comment by iSazonov** [comment]
> I personally use r2rdump utility

Thanks! If I remember right there was an env variable to show loaded dll-s at startup with their format.
